File: airbyte-connector-builder-server/src/main/java/io/airbyte/connector_builder/file_writer/AirbyteFileWriter.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connector_builder.file_writer;

import java.io.IOException;

/**
 * Handle temporary files containing information required for sending requests.
 */
public interface AirbyteFileWriter {

  /**
   * Writes `contents` to a temporary file prepended with `name`.
   */
  String write(final String name, final String contents) throws IOException;

  /**
   * Deletes files at `filepath`.
   */
  void delete(final String filepath);

}

File: airbyte-connector-builder-server/src/main/kotlin/io/airbyte/connectorbuilder/filewriter/AirbyteFileWriter.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.filewriter

import java.io.IOException

/**
 * Handle temporary files containing information required for sending requests.
 */
interface AirbyteFileWriter {
  /**
   * Writes `contents` to a temporary file prepended with `name`.
   */
  @Throws(IOException::class)
  fun write(
    name: String,
    contents: String,
  ): String

  /**
   * Deletes files at `filepath`.
   */
  fun delete(filepath: String)
}
